examples/io_write.rs

```diff
@@ -0,0 +1,40 @@
+use prometheus_client::{encoding::text::encode, metrics::counter::Counter, registry::Registry};
+use std::io::Write;
+
+/// Example showing how one could write to a file or socket instead of a string.
+/// For large metrics registries this will be more memory efficient
+fn main() {
+    let mut registry = <Registry>::with_prefix("stream");
+    let request_counter: Counter<u64> = Default::default();
+
+    registry.register(
+        "requests",
+        "How many requests the application has received",
+        request_counter.clone(),
+    );
+
+    let mut buf = String::new();
+    encode(&mut buf, &registry).unwrap();
+
+    let mut file = Vec::new();
+    let mut writer = IoWriterWrapper(&mut file);
+    encode(&mut writer, &registry).unwrap();
+
+    assert!(buf.as_bytes() == file);
+}
+
+pub struct IoWriterWrapper<W>(W)
+where
+    W: Write;
```

Comment:
```suggestion
pub struct IoWriterWrapper<W>(W);
```

Is the bound needed in the struct definition? It should only be needed in the `std::fmt::Write` implementation below.